sdk/dotnet/Pulumi/Deployment/GrpcMonitor.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Grpc.Core;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    class GrpcMonitor : IMonitor
```

Comment:
```suggestion
    internal class GrpcMonitor : IMonitor
```

Reply:
i'd almost prefer this as just having a generic `IRpc` interface which encapsulates all the rpc operations we need the deployment to perform.  Then we have our `GrpcRpc` impl of that interface.  That would read a lot better to me than calling this a `Monitor` which then talks to an `Engine`.